CRM/Event/Form/Registration/Confirm.php

```diff
@@ -547,13 +547,12 @@ public function postProcess() {
           $value['participant_status_id'] = $value['participant_status'] = array_search('Awaiting approval', $waitingStatuses);
         }
 
-        //there might be case user seleted pay later and
+        //there might be case user selected pay later and
         //now becomes part of run time waiting list.
         $value['is_pay_later'] = FALSE;
       }
-
-      // required only if paid event
-      if ($this->_values['event']['is_monetary'] && !($this->_allowWaitlist || $this->_requireApproval)) {
+      elseif ($this->_values['event']['is_monetary']) {
```

Comment:
Agree this looks equivalent. 👍 

* The removed conditional is the logical opposite of the previous `if ($this->_allowWaitlist || $this->_requireApproval) {`
* In the body of the `if { ... }`, there doesn't appear to be anything which could change the value of `$this->_allowWaitlist` or `$this->_requireApproval`.